src/commands/registerCommands.ts

```diff
@@ -181,20 +182,40 @@ async function grammarAssociationCommand(documentURI: Uri, languageClient: Langu
     // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
     const identifier = TextDocumentIdentifier.create(documentURI.toString());
     const grammarURI = fileUri[0];
+    const grammarFilename = path.basename(grammarURI.toString())
+    const currentFilename = path.basename(identifier.uri);
     try {
-      const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
-      // Insert the proper syntax for binding
-      const lspTextDocumentEdit = <TextDocumentEdit>result;
-      const workEdits = new WorkspaceEdit();
-      for (const edit of lspTextDocumentEdit.edits) {
-        workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
+      if (bindingType == `fileAssociation`) {
+        await commands.executeCommand(ClientCommandConstants.ASSOCIATE_GRAMMAR_FILE, currentFilename, grammarFilename);
```

Comment:
Replace executeCommand with a simple `function addToFileAssociation(currentFilename: string, grammarFilename: string)`. You will be sure that you set string as parameter (and not an Uri for instance). And we don't need to expose `binding with file association` as a command. We need just one entry point `Binding with schema/grammar`

Reply:
In the future, do you think we will want a command just for grammar association, or do you think we will only perform grammar association through the binding wizard and manually editing the settings.json?

Reply:
No I dont think. I think its good to have few commands. We will see if we have some user issues about that